spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/AbstractConnectionFactory.java

```diff
@@ -595,8 +595,10 @@ else if (soTimeout > 0) {
 								logger.warn("Timing out TcpNioConnection " +
 										    connection.getConnectionId());
 							}
-							connection.publishConnectionExceptionEvent(new SocketTimeoutException("Timing out connection"));
+							SocketTimeoutException exception = new SocketTimeoutException("Timing out connection");
+							connection.publishConnectionExceptionEvent(exception);
 							connection.timeout();
+							connection.sendExceptionToListener(exception);
```

Comment:
@garyrussell , wouldn't we have some issue that we `sendExceptionToListener` before `timeout()` ?

Thank for the explanation in advance!